torchqdynamics/mesolve.py

```diff
@@ -3,46 +3,90 @@
 
 import torch
 import torch.nn as nn
+from torch import Tensor
 
 from .odeint import AdjointQSolver, odeint
 from .solver import Rouchon, SolverOption
 from .solver_utils import inv_sqrtm, kraus_map
-from .types import TensorLike, TimeDependentOperator
+from .types import TensorLike, TensorTD
```

Comment:
So I actually thought a long time on this (wayyyy too long), and I'm not quite convinced about my solution. IMO there's three viable solutions right now (also seeing your other PR https://github.com/PierreGuilmin/torchqdynamics/pull/31):
 - `TDTensor`: it's concise and makes it clear it's a tensor underlying, but the user might not know what TD stands for (but maybe it's fine)
 - `TDOperator`: it's concise, and we can group with QObjs as well, and it differentiates with regular tensors.
 - `TimeDependentOperator`: it's very explicit, but can also be a bit scary when seeing for the first time. Also it implies that it's necessarely time dependent, when it could actually be time-indep.

I'll let you guys choose. What's your opinion @abocquet @eliegenois ?

Reply:
I'm down for `TDOperator`, and we can maybe then rename all the `TDOperatorLike` and `OperatorLike` to just `TDOperator` and `Operator` (let's do this in a subsequent PR if we choose to).